### PR TITLE
icicle-make: support for compiling icicle dictionaries to a standalone C program

### PR DESCRIPTION
**Pull request by jystic**
The C program reads from `stdin` and writes to `stdout`.

I'm not sure about the name, open to suggestions.

Depends on https://github.com/ambiata/x/pull/110

! @amosr @tranma

### CODE REVIEW

File: data/make/main.c [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    /* outputs */
+    const char *error;
+    iint_t fact_count;
+    iint_t entity_count;
+
+    /* stuff */
+    size_t facts_limit;   /* maximum number of facts per entity-attribute */
+    ibool_t has_drop; /* whether to write discarded facts to output or drop log */
+} psv_config_t;
```
Comment (jystic) [inline data/make/main.c]: This data structure is duplicated from one of the `data/sea/*.h` files. Not sure of a way to de-dupe this that isn't overly complex :(

**Comment by jystic** [comment]
Next stop ASAN! https://github.com/ambiata/icicle/issues/452 /cc @erikd-ambiata

File: icicle-compiler/main/icicle-make.hs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+tombstonesOfDictionary =
+  let
+    takeEntry = \case
+      DictionaryEntry a (ConcreteDefinition _ ts _) _ ->
+        [(a, ts)]
+      _ ->
+        []
+
+  in
+    Map.fromList . concatMap takeEntry . dictionaryEntries
```
Comment (tranma) [inline icicle-compiler/main/icicle-make.hs]: I think this already exists somewhere....

File: icicle-compiler/main/icicle-make.hs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+renderArgs :: [Text] -> Text
+renderArgs =
+  let
+    render opt =
+      if " " `T.isInfixOf` opt then
+        "\"" <> opt <> "\""
+      else
+        opt
```
Comment (tranma) [inline icicle-compiler/main/icicle-make.hs]: 👍

**Comment by tranma** [comment]
this is great 🍨

File: icicle-compiler/main/icicle-make.hs (hunk 2 above)
Comment (jystic) [inline icicle-compiler/main/icicle-make.hs]: will consolidate

**Pull request merged**